model/src/main/java/com/basistech/rosette/dm/MapAttribute.java

```diff
@@ -0,0 +1,217 @@
+/*
+* Copyright 2018 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.basistech.rosette.dm;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Maps;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A container for a keyed collection of attributes of a type.
+ * Like all other attributes, it stores extended properties.
+ * It is immutable and throws for attempts to use methods that would modify it.
+ *
+ * @param <K> The type of the keys in the map.
+ * @param <V> The type of the values in the map.
+ */
+public class MapAttribute<K, V extends BaseAttribute> extends BaseAttribute implements Map<K, V>, Serializable {
+    private static final long serialVersionUID = 222L;
```

Comment:
250 for the current convention, I think, assuming that's the version.
```suggestion
    private static final long serialVersionUID = 250L;
```

Reply:
We don't have and serialized/stored objects or calling via RMI, so this is not strictly needed. As a convention, we could say any time you touch a class/field/method you should also update the number to a different one, or you can generated it using `serialver` on the compiled class, or [configure IDEA](https://stackoverflow.com/questions/24573643/how-to-generate-serial-version-uid-in-intellij) to do it.